service/config_provider.go

```diff
@@ -75,6 +75,37 @@ type configProvider struct {
 	watcher chan error
 }
 
+// ConfigProviderOptions has options that change the behavior of ConfigProvider
+// returned by NewConfigProvider()
+type ConfigProviderOptions struct {
+	configMapProviders map[string]configmapprovider.Provider
+	cfgMapConverters   []config.MapConverterFunc
+	configUnmarshaler  configunmarshaler.ConfigUnmarshaler
+}
```

Comment:
this type is not needed. You can change the option to be: `type ConfigProviderOption func(opts *configProvider)`. External user cannot and should not be able to extend the `ConfigProviderOption` anyway.